Makefile

```diff
@@ -3,9 +3,9 @@ export BASE_DIR := $(shell dirname $(realpath $(lastword $(MAKEFILE_LIST))))
 # Build flags
 export CXX := g++
 USER_CXXFLAGS :=
-HOST_CXXFLAGS := -O2 -fPIC -fdiagnostics-show-option -felide-constructors -fmessage-length=0 -fno-math-errno -ftree-vectorize -fvisibility-inlines-hidden --param vect-max-version-for-alias-checks=50 -msse3 -pipe -pthread -Xassembler --compress-debug-sections -Werror=address -Wall -Werror=array-bounds -Wno-attributes -Werror=conversion-null -Werror=delete-non-virtual-dtor -Wno-deprecated -Werror=format-contains-nul -Werror=format -Wno-long-long -Werror=main -Werror=missing-braces -Werror=narrowing -Wno-non-template-friend -Wnon-virtual-dtor -Werror=overflow -Werror=overlength-strings -Wparentheses -Werror=pointer-arith -Wno-psabi -Werror=reorder -Werror=return-local-addr -Wreturn-type -Werror=return-type -Werror=sign-compare -Werror=strict-aliasing -Wstrict-overflow -Werror=switch -Werror=type-limits -Wunused -Werror=unused-but-set-variable -Wno-unused-local-typedefs -Werror=unused-value -Wno-error=unused-variable -Wno-vla -Werror=write-strings
+HOST_CXXFLAGS := -O2 -g -fPIC -fdiagnostics-show-option -felide-constructors -fmessage-length=0 -fno-math-errno -ftree-vectorize -fvisibility-inlines-hidden -msse3 -pipe -pthread -Werror=address -Wall -Werror=array-bounds -Wno-attributes -Werror=conversion-null -Werror=delete-non-virtual-dtor -Wno-deprecated -Werror=format -Wno-long-long -Werror=main -Werror=missing-braces -Werror=narrowing -Wnon-virtual-dtor -Werror=overflow -Werror=overlength-strings -Wparentheses -Werror=pointer-arith -Wno-reorder-ctor -Wreturn-type -Werror=return-type -Wno-sign-compare -Werror=strict-aliasing -Wstrict-overflow -Werror=switch -Werror=type-limits -Wunused -Wno-unused-local-typedefs -Wno-unused-function -Werror=unused-value -Wno-error=unused-variable -Wno-vla -Werror=write-strings
```

Comment:
Well, I added `-g` to help debug things :-)
Do you prefer that we add it by hand / from the command line when needed ?

I removed `-Xassembler --compress-debug-sections` because compressed debug sections do not seem to be supported by `dpcpp`:
```
/opt/intel/oneapi/compiler/latest/linux/bin/dpcpp -fsycl -I/opt/intel/oneapi/dpcpp-ct/latest/include /data/user/fwyzard/pixeltrack-standalone/obj/sycltest/DataFormats/FEDRawDataCollection.cc.o /data/user/fwyzard/pixeltrack-standalone/obj/sycltest/DataFormats/FEDHeader.cc.o /data/user/fwyzard/pixeltrack-standalone/obj/sycltest/DataFormats/FEDNumbering.cc.o /data/user/fwyzard/pixeltrack-standalone/obj/sycltest/DataFormats/FEDRawData.cc.o /data/user/fwyzard/pixeltrack-standalone/obj/sycltest/DataFormats/FEDTrailer.cc.o -O2 -g -fPIC -pthread -Wl,-E -lstdc++fs -shared -Wl,-z,defs -L/data/user/fwyzard/pixeltrack-standalone/lib/sycltest  -L/data/user/fwyzard/pixeltrack-standalone/external/tbb/lib -ltbb  -o /data/user/fwyzard/pixeltrack-standalone/lib/sycltest/libDataFormats.so
/usr/bin/ld: /tmp/fwyzard/FEDRawDataCollection-d99872.o: unable to initialize decompress status for section .debug_loc
/usr/bin/ld: /tmp/fwyzard/FEDRawDataCollection-d99872.o: unable to initialize decompress status for section .debug_loc
/tmp/fwyzard/FEDRawDataCollection-d99872.o: file not recognized: File format not recognized
dpcpp: error: linker command failed with exit code 1 (use -v to see invocation)
```

By the way, compressed debug sections are also not supported by `cuda-gdb`, so I would really be in favour of removing this option.


Other options are not understood:
```
dpcpp: warning: argument unused during compilation: '--param vect-max-version-for-alias-checks=50' [-Wunused-command-line-argument]
warning: unknown warning option '-Werror=format-contains-nul' [-Wunknown-warning-option]
warning: unknown warning option '-Wno-non-template-friend'; did you mean '-Wno-unsupported-friend'? [-Wunknown-warning-option]
warning: unknown warning option '-Werror=return-local-addr'; did you mean '-Werror=return-stack-address'? [-Wunknown-warning-option]
warning: unknown warning option '-Werror=unused-but-set-variable'; did you mean '-Werror=unused-const-variable'? [-Wunknown-warning-option]
```
I think thse might be options we switch off for the clang builds in CMSSW as well ?

Some more were giving a lot of positives in the `dpct` headers.
Since I managed to get rid of them after the migration, they can be added back.

Reply:
> Do you prefer that we add it by hand / from the command line when needed ?

Yes (essentially to mimic the behavior of CMSSW)

> By the way, compressed debug sections are also not supported by `cuda-gdb`, so I would really be in favour of removing this option.

Ok, let's keep it removed.

For the remaining options, I'd suggest to leave them in `CXXFLAGS` , but define later in the `Makefile` e.g.
```Makefile
export SYCL_CXXFLAGS := $(filter-out option1 option2,$(CXXFLAGS))
```
and replace `$(CXXFLAGS)` with `$(SYCL_CXXFLAGS)` in `src/sycltest/Makefile`. Then it would be self-documented which options (that we pass to gcc) need to be removed (even if they are likely removed from CMSSW clang builds as well).

Reply:
OK, I've done something like what you suggested.